### DISCUSSION TITLE
NGXS Schematics

### DISCUSSION THREAD

**Discussion by markwhitfeld** [ngxs/store#2053] [RFC & Proposals]
We can add:
- Standalone API support (update the root providers)
- Option to add a state at the root or the feature (router) level
- Add a plugin
  - Should offer options of what to install, excluding those already installed (should mention what is installed in a message)
  - Should add the package and add it to the root provider with the default options
- Migrations (this needs to be covered in a separate RFC)

Please add proposals any other proposals for new schematics or changes that should be made as a comment below.

**Comment by Akshat111111** [comment]
> We can add:
> 
> * Standalone API support (update the root providers)
> * Option to add a state at the root or the feature (router) level
> * Add a plugin
>   
>   * Should offer options of what to install, excluding those already installed (should mention what is installed in a message)
>   * Should add the package and add it to the root provider with the default options
> * Migrations (this needs to be covered in a separate RFC)
> 
> Please add proposals any other proposals for new schematics or changes that should be made as a comment below.

I totally agree with your proposal, but apart from it I also suggest Template rendering ensuring that complex templates don't slow down the CLI during code generation.The focus should also be in Implementing intelligent dependency management in order to minimize the startup time.

Apart  from it,we can also  add Package automatically to the root provider by updating the app.module.ts or some equivalent.
